tools/generate-achievement-types.js

```diff
@@ -0,0 +1,28 @@
+const { readFileSync, writeFileSync } = require("fs");
+const { resolve } = require("path");
+
+const rawData = readFileSync(resolve(__dirname, "../src/Achievements/AchievementData.json"), "utf-8");
```

Comment:
I think you can avoid needing to write this generator script. Instead, I think you can make use of import attributes: <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import/with>

Note that that feature is not widely available, *but* Webpack will polyfill it for us, essentially embedding the result, so there should be no runtime issues. The only question is if there are *build-time* issues that prevent it from working as the typed solution you want here.